alerting-config-service-impl/src/main/java/org/hypertrace/alerting/config/service/EventConditionConfigServiceRequestValidator.java

```diff
@@ -0,0 +1,43 @@
+package org.hypertrace.alerting.config.service;
+
+import static org.hypertrace.config.validation.GrpcValidatorUtils.validateRequestContextOrThrow;
+
+import com.google.common.base.Preconditions;
+import org.hypertrace.alerting.config.service.v1.CreateEventConditionRequest;
+import org.hypertrace.alerting.config.service.v1.DeleteEventConditionRequest;
+import org.hypertrace.alerting.config.service.v1.GetAllEventConditionsRequest;
+import org.hypertrace.alerting.config.service.v1.UpdateEventConditionRequest;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+
+public class EventConditionConfigServiceRequestValidator {
+
+  public void validateCreateEventConditionRequest(
+      RequestContext requestContext, CreateEventConditionRequest request) {
+    validateRequestContextOrThrow(requestContext);
+    Preconditions.checkArgument(request.hasNewEventCondition(), "EventCondition should be present");
+    Preconditions.checkArgument(
+        request.getNewEventCondition().getEventConditionData().hasMetricAnomalyEventCondition(),
+        "MetricAnomalyEventCondition should be present");
+  }
+
+  public void validateUpdateEventConditionRequest(
+      RequestContext requestContext, UpdateEventConditionRequest request) {
+    validateRequestContextOrThrow(requestContext);
+    Preconditions.checkArgument(request.hasEventCondition(), "EventCondition should be present");
+    Preconditions.checkArgument(
```

Comment:
Like delete, shouldn't the update check for id argument as well - `!request.getEventConditionId().isBlank()`?